macros/src/types/enum.rs

```diff
@@ -81,17 +82,17 @@ fn format_variant(
     }
 
     let untagged_variant = variant_attr.untagged;
-    let name = match (variant_attr.rename.clone(), &enum_attr.rename_all) {
+    let ts_name = match (variant_attr.rename.clone(), &enum_attr.rename_all) {
         (Some(rn), _) => rn,
-        (None, None) => variant.ident.to_string(),
-        (None, Some(rn)) => rn.apply(&variant.ident.to_string()),
+        (None, None) => make_string_literal(&variant.ident.to_string(), variant.ident.span()),
+        (None, Some(rn)) => make_string_literal(&rn.apply(&variant.ident.to_string()), variant.ident.span()),
```

Comment:
Since the rest of the code now expects an `Expr`, there are a couple of places where we construct an `Expr` from a string